Source/JavaScriptCore/llint/WebAssembly.asm

```diff
@@ -640,54 +656,57 @@ end
             emit "movz x5, #0xBAD"
             emit "movz x6, #0xBAD"
             emit "movz x7, #0xBAD"
-            emit "movz x8, #0xBAD"
         end
     end
 
     clobberVolatileRegisters()
 
+
 if ARM64 or ARM64E or ARMv7
-    const JSEntrypointInterpreterCalleeSaveSpaceStackAligned = (4 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
+    const JSEntrypointInterpreterSpillSpaceStackAligned = (4 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
 else
-    const JSEntrypointInterpreterCalleeSaveSpaceStackAligned = (8 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
+    const JSEntrypointInterpreterSpillSpaceStackAligned = (8 * SlotSize + StackAlignment - 1) & ~StackAlignmentMask
 end
 
     macro saveJSEntrypointInterpreterRegisters()
-        subp JSEntrypointInterpreterCalleeSaveSpaceStackAligned, sp
+        subp JSEntrypointInterpreterSpillSpaceStackAligned, sp
         if ARM64 or ARM64E
-            storepairq boundsCheckingSize, metadataTable, -16[cfr]
-            storepairq wasmInstance, memoryBase, -32[cfr]
+            storepairq memoryBase, boundsCheckingSize, -2 * SlotSize[cfr]
+            storep wasmInstance, -3 * SlotSize[cfr]
+            # These are just for us to spill things
+            # MPCSpill
         elsif X86_64
             # These must match the wasmToJS thunk, since the unwinder won't be able to tell the difference between us and them.
             # See calleeSaveRegistersImpl
-            storep metadataTable, -0x8[cfr]
-            storep boundsCheckingSize, -0x10[cfr]
-            storep memoryBase, -0x18[cfr]
-            storep wasmInstance, -0x20[cfr]
+            storep boundsCheckingSize, -1 * SlotSize[cfr]
+            storep memoryBase, -2 * SlotSize[cfr]
+            storep wasmInstance, -3 * SlotSize[cfr]
             # This gets restored before call too, since it doesn't get saved/restored by the wasmToJS thunk
+            storep MPC, MPCCSRSlot[cfr]
             storep Accumulator, AccumulatorCSRSlot[cfr]
             # These are just for us to spill things
             # AccumulatorSpill
-            # CP
+            # MPCSpill
+            # CalleeExecutableAddress
         else
             error
         end
     end
 
     macro restoreJSEntrypointInterpreterRegisters()
         if ARM64 or ARM64E
-            loadpairq -16[cfr], boundsCheckingSize, metadataTable
-            loadpairq -32[cfr], wasmInstance, memoryBase
+            loadpairq -2 * SlotSize[cfr], memoryBase, boundsCheckingSize
+            loadpairq -4 * SlotSize[cfr], ws0, wasmInstance
```

Comment:
Nit: It's a little weird this does a loadpair but we do a single store above.

Reply:
Why is this weird?  This seems appropriate for ARM64.  What am I missing?

Reply:
Oh, the extra load into ws0.  That looks unneeded.

Reply:
Nice catch, I missed this one. Thanks!